### DISCUSSION TITLE
How to get the estimated time it takes to train a yolov8 model

### DISCUSSION THREAD

**Discussion by ssunyoung2** [ultralytics/ultralytics#11746] [Q&A]
After training starts, can I know the time from when 1 epoch has not progressed until it is completed?

The final goal is to obtain the expected training time up to 100 epochs from the training run.

**Comment by glenn-jocher** [comment]
@ssunyoung2 yes, you can estimate the training time per epoch and then calculate the total time for a desired number of epochs.

When you begin training a YOLOv8 model, the time taken for each epoch is displayed in the CLI or Python output. After the first epoch completes, you'll see an estimate of the remaining time that updates at the end of each epoch.

To get a baseline estimation for up to 100 epochs, you can simply multiply the average time per epoch by the number of epochs. Here's a Python example using YOLO from Ultralytics:

```python
from ultralytics import YOLO

# Load and train model
model = YOLO('yolov8n.yaml')  # adjust the model configuration as needed
results = model.train(data='coco128.yaml', epochs=100, imgsz=640)

# Results contain the time taken for each epoch
time_per_epoch = results.time  # average time per epoch in seconds
estimated_total_time = time_per_epoch * 100  # for 100 epochs
print(f"Estimated Total Time for 100 Epochs: {estimated_total_time / 3600:.2f} hours")
```

This will give you a rough estimate based on the training progress. Keep in mind runtime can vary based on hardware specifications, dataset complexity, and model architecture adjustments.
